evaluate_data_frame REST test can fail from unexpected warnings

**Build scan**: https://gradle-enterprise.elastic.co/s/vptinxha2wxme

**Repro line**:
```
./gradlew ':x-pack:plugin:yamlRestTest' --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=ml/evaluate_data_frame/Test classification given missing actual_field}" \
  -Dtests.seed=96A8B5076DF8D87B \
  -Dtests.locale=it \
  -Dtests.timezone=America/Guatemala \
  -Druntime.java=8
```

**Reproduces locally?**: no

**Applicable branches**: 7.13, likely others

**Failure history**: only one failure of its type so far

**Failure excerpt**:
```
java.lang.AssertionError: Failure at [ml/evaluate_data_frame:189]: got unexpected warning header [
    	299 Elasticsearch-7.13.0-SNAPSHOT-45067c011e99363d974ebf96244bec212663eada "this request accesses system indices: [.ml-config], but in a future major version, direct access to system indices will be prevented by default"
    ]
        at __randomizedtesting.SeedInfo.seed([96A8B5076DF8D87B:1EFC8ADDC304B583]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:451)
```

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/evaluate_data_frame.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/evaluate_data_frame.yml
@@ -187,7 +187,8 @@ setup:
           }
 
   - do:
-      indices.refresh: {}
+      indices.refresh:
+        index: utopia
 
 ---
 "Test outlier_detection auc_roc":

